ucp/_libs/packed_remote_key.pyx

```diff
@@ -56,7 +56,7 @@ cdef class PackedRemoteKey:
         packed_key = PackedRemoteKey(<uintptr_t>key, len)
         ucp_rkey_buffer_release(key)
         assert_ucs_status(status)
-        return PackedRemoteKey(<uintptr_t>key, len)
+        return packed_key
```

Comment:
Good catch, but what are we actually returning here? It seems like `key` is released by `ucp_rkey_buffer_release`, so I'm not sure what exactly is that `packed_key` contains.

Reply:
It's a new instance of the `PackedRemoteKey` class. The class makes it's own copy of the key since it can either from from UCX (thus needs to be freed with UCX functions) or from the external runtime (thus needs to be managed and freed as a bytes/buffer routines). Though now that I'm thinking about the problem and scaling it, a better solution would be to just do the copy/buffer stuff in the memory handle case and then not make a copy in the buffer case. Hmm... add one to the todo file.

Reply:
Makes sense, it's indeed a bit misleading to read just this function without looking at the constructor, but for now this will do. Thanks Matt!